phpBB/phpbb/textformatter/s9e/utils.php

```diff
@@ -35,6 +35,43 @@ public function clean_formatting($xml)
 	}
 
 	/**
+	* Return given string between quotes
+	*
+	* Will use either single- or double- quotes depending on whichever requires to be escaped.
+	* Quotes and backslashes are escaped with backslashes where necessary
+	*
+	* @param  string $str Original string
+	* @return string      Escaped string within quotes
+	*/
+	protected function enquote($str)
+	{
+		$quote = (strpos($str, '"') === false || strpos($str, "'") !== false) ? '"' : "'";
```

Comment:
and what if the string contains both quotes? you will generate `"simple-quote ' double-quote " end of string"`?